oauthproxy.go

```diff
@@ -976,9 +976,7 @@ func (p *OAuthProxy) addHeadersForProxying(rw http.ResponseWriter, req *http.Req
 		}
 
 		if len(session.Groups) > 0 {
-			for _, group := range session.Groups {
-				rw.Header().Add("X-Auth-Request-Groups", group)
-			}
+			rw.Header().Set("X-Auth-Request-Groups", strings.Join(session.Groups,","))
```

Comment:
We would need to make this change to `X-Forwarded-Groups` as well.

@JoelSpeed I remember this PR used to have the groups joined explicitly into 1 string with `,` and you advised for `Add` instead.  Do you have more insight to the nuance of this and how these settings are called when the actual `ResponseWriter` is rendering these headers?

I see this distinction between `Get` vs `Values` on array headers, but not quickly finding how `ResponseWriter` renders:

```
// Get gets the first value associated with the given key.
// It is case insensitive; CanonicalMIMEHeaderKey is used
// to canonicalize the provided key.
// If there are no values associated with the key, Get returns "".
// To use non-canonical keys, access the map directly.
func (h MIMEHeader) Get(key string) string {
	if h == nil {
		return ""
	}
	v := h[CanonicalMIMEHeaderKey(key)]
	if len(v) == 0 {
		return ""
	}
	return v[0]
}

// Values returns all values associated with the given key.
// It is case insensitive; CanonicalMIMEHeaderKey is
// used to canonicalize the provided key. To use non-canonical
// keys, access the map directly.
// The returned slice is not a copy.
func (h MIMEHeader) Values(key string) []string {
	if h == nil {
		return nil
	}
	return h[CanonicalMIMEHeaderKey(key)]
}
```

Reply:
If only the first entry is ending up in the actual header, seems like the `http.ResponseWriter` is using `Get` where `X-Forwarded-Groups` that still had all the entries (but this was a `http.Request`) might be using `Values`...?

Reply:
Yeah. I only saw the first entry and  I have the same thought here. But I couldn't drill down to the exact line where Get is called. Also updated the X-Forwarded-Groups